src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1954,17 +1956,38 @@ static Set<String> expandFileNames(String[] filePathList, String queryId)
           result.add(((java.io.File) file).getCanonicalPath());
         }
       } catch (Exception ex) {
-        throw new SnowflakeSQLException(
-            queryId,
-            ex,
-            SqlState.DATA_EXCEPTION,
-            ErrorCode.FAIL_LIST_FILES.getMessageCode(),
-            "Exception: "
-                + ex.getMessage()
-                + ", Dir="
-                + entry.getKey()
-                + ", Patterns="
-                + entry.getValue().toString());
+        if (ex instanceof UncheckedIOException) {
+          if (!dir.exists()) {
```

Comment:
It's thrown by [this](https://github.com/snowflakedb/snowflake-jdbc/blob/b765408004ccf44efa325260ef07b72b49841c28/src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java#L1955) line so it should be initialized but I'll add a null check just in case.